src/CppUTest/CommandLineTestRunner.cpp

```diff
@@ -51,21 +49,44 @@ int CommandLineTestRunner::RunAllTests(int ac, char** av)
 int CommandLineTestRunner::RunAllTests(int ac, const char** av)
 {
     int result = 0;
-    ConsoleTestOutput output;
+    TestRegistry* registry = TestRegistry::getCurrentRegistry();
+    CommandLineArguments arguments( ac, av );
+    arguments.parse( registry->getFirstPlugin() );
+    TestOutput* output;
+    if( arguments.isJUnitOutput() ) {
+        JUnitTestOutput* jUnitOutput = new JUnitTestOutput;
+        jUnitOutput->setPackageName( arguments.getPackageName() );
+        output = jUnitOutput;
+    }
+    else {
+        output = new ConsoleTestOutput;
+    }
 
     MemoryLeakWarningPlugin memLeakWarn(DEF_PLUGIN_MEM_LEAK);
     memLeakWarn.destroyGlobalDetectorAndTurnOffMemoryLeakDetectionInDestructor(true);
-    TestRegistry::getCurrentRegistry()->installPlugin(&memLeakWarn);
+    registry->installPlugin( &memLeakWarn );
+
+    SetPointerPlugin pPlugin( DEF_PLUGIN_SET_POINTER );
+    registry->installPlugin( &pPlugin );
 
+    if(arguments.isSuccesfullyParsed())
     {
-        CommandLineTestRunner runner(ac, av, &output, TestRegistry::getCurrentRegistry());
+        CommandLineTestRunner runner( output, registry, &arguments );
         result = runner.runAllTestsMain();
     }
+    else {
+        output->print( arguments.usage() );
+    }
+
+    registry->removePluginByName( DEF_PLUGIN_SET_POINTER );
 
     if (result == 0) {
-        output << memLeakWarn.FinalReport(0);
+        *output << memLeakWarn.FinalReport(0);
     }
-    TestRegistry::getCurrentRegistry()->removePluginByName(DEF_PLUGIN_MEM_LEAK);
+    registry->removePluginByName( DEF_PLUGIN_MEM_LEAK );
+
+    delete output;
+
```

Comment:
The reason why the original code was the way it was is that we want *all* dynamic memory allocation to be within the memLeakWarn and FinalReport to be done. That is why the original ConsoleTestOuput was created as a stack object, so that it doesn't allocate memory and doesn't interfere with the memory leak checking.

So, the design here will probably need to keep the RunAllTests to be the same (minimal) while being able to change the TestOutput.